pkg/appfile/parser.go

```diff
@@ -238,6 +238,38 @@ func (p *Parser) parseTrait(ctx context.Context, name string, properties map[str
 	}, nil
 }
 
+// ValidateComponentNames validate all component name whether repeat in cluster and template
+func (p *Parser) ValidateComponentNames(ctx context.Context, af *Appfile) (int, error) {
+	existCompNames := make(map[string]string)
+	existApps := v1beta1.ApplicationList{}
+	if err := p.client.List(ctx, &existApps); err != nil {
+		return 0, err
```

Comment:
ignore not exist？

Reply:
existApps will empty when not exist, this err may connect apiserver error or list permission denied